Pod/Classes/ios/NYTPhotosOverlayView.m

```diff
@@ -49,6 +50,10 @@ - (void)layoutSubviews {
     }];
     
     [super layoutSubviews];
+
+    if ([self.captionView respondsToSelector:@selector(setPreferredMaxLayoutWidth:)]) {
```

Comment:
I'm not sure. Here are arguments for and against changing to `conformsToProtocol:`

#### for

- the current code carries implicit knowledge of what's in the `NYTPhotoCaptionViewLayoutWidthHinting` protocol, spilled across 2 lines (though a change to the protocol would still cause a compiler warning)
- we should check exactly what we're casting to, before casting

#### against

- what we really care about is whether we can set `preferredMaxLayoutWidth`, and with this approach certain classes (`UILabel`) get correct behavior for free. (the cast is practically a detail to satisfy the compiler, though a change to the protocol would still cause a compiler warning)
- `conformsToProtocol:` requires a global runtime lock; `respondsToSelector:` does not
- even if we check that it conforms to the protocol, the code still carries implicit knowledge about what methods are in the protocol because we still have to cast before sending the message